static/app/views/teamInsights/overview.tsx

```diff
@@ -56,11 +57,17 @@ function TeamInsightsOverview({
   router,
 }: Props) {
   const query = location?.query ?? {};
-  const currentTeamId = query.team ?? teams[0]?.id;
+  const localStorageKey = `teamInsightsSelectedTeamId:${organization.slug}`;
+  let localTeamId = localStorage.getItem(localStorageKey);
+  if (localTeamId && !teams.find(team => team.id === localTeamId)) {
+    localTeamId = null;
+  }
+  const currentTeamId = query.team ?? localTeamId ?? teams[0]?.id;
```

Comment:
Is it possible that `query.team` will not be in the `projects` array? If yes, should we handle that?